### PR TITLE
fix error in #2 by @IcedOtaku

### PR DESCRIPTION
**Pull request by IcedOtaku**

### CODE REVIEW

**Comment by IcedOtaku** [comment]
I rewrote this js logic using python.
See [hosts-py](https://github.com/IcedOtaku/hosts-py).

**Comment by ineo6** [comment]
Update check is still needed. Or our commits history will be too huge.

**Comment by IcedOtaku** [comment]
> Update check is still needed. Or our commits history will be too huge.

Thank you for your reply.
But maybe you should check this [commit](https://github.com/ineo6/hosts/commit/60f77512e6d09cd5904a2e6062f351c5f6f23492), and there are more commits like this.
The update logic always generate a commit even if the hosts were not changed at all, isn't it ?

**Comment by ineo6** [comment]
You are right. There is a bug about it. We should only compare hosts content.

So we can remove those code comment first and then to compare.

**Comment by IcedOtaku** [comment]
> You are right. There is a bug about it. We should only compare hosts content.
> 
> So we can remove those code comment first and then to compare.

I agree. Please let me try to fix this and make this PR work.

**Comment by IcedOtaku** [comment]
Hi, recently I optimized the updating logic by adjusting reg rule [here](https://github.com/IcedOtaku/hosts/blob/master/index.js#L129).👷!
@ineo6

**Comment by ineo6** [comment]
Great! Let me check.

File: index.js [hunk 1]
```diff
@@ -87,10 +82,13 @@ async function findIp(host) {
 
   $('#dnsinfo>tr')
     .each(function (i, element) {
-      ipList.push($(this)
+      let td = $(this).children();
+      if ($(td[1]).text() == 'A') {
```
Comment (ineo6) [inline index.js:86]: This should use the type-safe equality operators === .

File: index.js [hunk 2]
```diff
@@ -125,13 +123,14 @@ function updateMd(content) {
...
-  }
 
-  const regMatch = /```bash([\s\S]*)```/.exec(prevMd);
+  }
 
+  const regMatch = prevMd.match(/GitHub Host Start([\s\S]*)# Update at/);
   const prevHost = regMatch ? regMatch[1].trim() : '';
 
   const needUpdate = prevHost !== content;
+  console.log("needUpdate", needUpdate);
```
Comment (ineo6) [inline index.js:133]: Remove console.

**Review by ineo6 — COMMENTED** [review]
Todo list.

**Comment by ineo6** [comment]
Merge soon.

**Pull request merged**